syntax = "proto3";

package transport_catalogue_pb;

import "map_renderer.proto";

import "transport_router.proto";

message SerializerSettings {
   string  file = 1;
}

message Coordinates {
  double lat = 1;
  double lng = 2;
}

message Distance_to_stop {
   string  stop_name = 1;
  uint32 distance = 2;
}

message Stop {
   string  name = 1;
  Coordinates coordinates = 2;
  repeated Distance_to_stop distances_to_stop = 3;
}

message Bus {
   string  name = 1;
  repeated  string  stops = 2;
  bool is_roundtrip = 3;
}

message RoutingSettings {
  double bus_wait_time = 1;
  double bus_velocity = 2;
};

message TransportCatalogue {
  repeated Stop stops = 1;
  repeated Bus buses = 2;
}

message TransportPackage {
  TransportCatalogue catalogue = 1;
  RenderSettings render_settings = 2;
  RoutingSettings routing_settings = 3;
  TransportRouter transport_router = 4;
}